### ISSUE TITLE
Incorrect C# client generation for polymorphic types

### ISSUE THREAD

**Issue by ben-wallis** [RicoSuter/NSwag#2078]
I've got a Swagger server hosted using Swashbuckle.AspNetCore v5.0.0-rc2 that contains a controller that has a polymorphic return type (specified using `oneOf` in the attached [swagger.json](https://github.com/RicoSuter/NSwag/files/3038698/swagger.txt)) and NSwagStudio seems to be generating incorrect non-compilable client methods using a generic type of "Anonymous" for one of the endpoints, and one of the sub-types instead of the base type for another endpoint.

This:

`public System.Threading.Tasks.Task<System.Collections.Generic.List<Anonymous>> SourcesGetAsync()` 

should be:

`public System.Threading.Tasks.Task<System.Collections.Generic.List<SourceDefinitionBaseDto>> SourcesGetAsync()`

and this:

`public System.Threading.Tasks.Task<ApplicationSourceDefinitionDto> SourcesGetAsync(System.Guid id)`

should be:

`public System.Threading.Tasks.Task<SourceDefinitionBaseDto> SourcesGetAsync(System.Guid id)`

The nswag config file used is [here](https://github.com/RicoSuter/NSwag/files/3038743/test.nswag.txt)

**Comment by ben-wallis** [comment]
I see that this is likely related to #13 / #839 - is this likely to be implemented soon?

**Comment by RicoSuter** [comment]
Are the correct issues linked?
Can you provide a sample swagger spec?

**Comment by ben-wallis** [comment]
Ah sorry, I didn't notice those were NJsonSchema issues rather than NSwagStudio issues, I've updated the links above.

**Comment by ben-wallis** [comment]
Update: This doesn't happen when using NSwag instead of Swashbuckle.AspNetCore to run the swagger endpoint so is potentially a Swashbuckle issue with generating the json schema.

**Comment by RicoSuter** [comment]
Yes, maybe. But maybe its an issue in nswag/njs which is just handled correct because its one toolchain... the goal is that nswag/njs also works with other tools.
